.github/workflows/build.yaml

```diff
@@ -35,3 +38,58 @@ jobs:
           tags: "ghcr.io/${{ github.repository }}:${{ github.sha }}"
           cache-from: type=gha,scope=${{ github.repository }}
           cache-to: type=gha,scope=${{ github.repository }},mode=max
+
+  trivy-scan-code:
+    needs: build
+    name: Trivy Vulnerability Scan for Code
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Aqua Security Trivy Scan
+        uses: aquasecurity/trivy-action@master
+        with:
+          scan-type: 'fs'
+          security-checks: 'vuln,secret,config'
+          ignore-unfixed: true
+          format: 'sarif'
+          output: 'trivy-results.sarif'
+
+      - name: Upload Trivy scan results to GitHub Security tab
+        uses: github/codeql-action/upload-sarif@v2
+        with:
+          sarif_file: 'trivy-results.sarif'
+          category: 'code'
+
+      - name: Image Scan ghcr
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: 'ghcr.io/${{ github.repository }}:${{ github.sha }}'
+          format: 'sarif'
+          output: 'trivy-image-results-ghcr.sarif'
+          exit-code: '1'
+          ignore-unfixed: true
+          vuln-type: 'os,library'
+
+      - name: Image Scan redis
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: 'redis:alpine'
+          format: 'sarif'
+          output: 'trivy-image-results-redis.sarif'
+          exit-code: '1'
+          ignore-unfixed: true
+          vuln-type: 'os,library'
+
+      - name: Upload Trivy Image Ghcr scan results to GitHub Security tab
+        uses: github/codeql-action/upload-sarif@v2
+        with:
+          sarif_file: 'trivy-image-results-ghcr.sarif'
+          category: 'image-ghcr'
```

Comment:
What do you consider as a category? The fact that it's a image or where it is located?

ghcr is a registry but it could be published elsewhere, should we change the category if we decide to change where we publish the image?

Reply:
It's used for metadata and reporting purposes, you can write whatever you want in it, it will not change anything

Reply:
It may not change stuff, but having extra metadata is super useful, so I'm glad you were able to find and make use of this option.

Now, I see you changed it as `image`. I want you to do a couple of research to see if `image` is the correct term. Ideally I would like us to use the `category` option everywhere and make use of the same term for each type of scans. (As we may find ourself filtering across multiple repo one day).

Reply:
Ok I see! I will double check and found the perfect term. Do you have in mind something like this : `<prefix>-<name_of_répo>` ?

Reply:
Finding the perfect term is an impossible task, 😂 so don't worry too much about it. Just some extra thought to ensure you are happy with the name.

I honestly think that `category: image` is good. Other options I can think of is `docker` or `container`. I don't have a strong opinion for now, use what you think fits. 

Don't include the name of the repo, as we want the category to be named the same across different repo, allowing to do searches like `org:scality category:image severity:critical` and get the list accordingly. 

The thing to remember here is every time you will add metadata on this security topic, try to give it some thought on how it could be used.

Reply:
Ohh ok you're right. I think `image` is a good term, if I find something else I will change